File: api-net/Controllers/SettingsController.cs
using LiftOtonom.Api.Data;
using LiftOtonom.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace LiftOtonom.Api.Controllers;

[ApiController]
[Authorize]
[Route("api/v1/settings")]
public class SettingsController(AppDbContext db) : ControllerBase
{
    public record SettingsDto(string? Name, string? Phone, string? Email, string? Address, string? TaxNumber, string? TaxOffice);

    [HttpGet]
    public async Task<IActionResult> Show()
    {
        var t = await db.Tenants.FindAsync(db.CurrentTenantId!.Value) ?? throw new ApiException(404, "Firma bulunamadı.");
        return Ok(new
        {
            t.Name, t.Slug, t.Phone, t.Email, t.Address, t.TaxNumber, t.TaxOffice, t.LogoUrl,
            t.Plan, t.PlanExpiresAt, t.SmsBalance,
        });
    }

    [HttpPut]
    [Authorize(Roles = "manager")]
    public async Task<IActionResult> Update(SettingsDto dto)
    {
        var t = await db.Tenants.FindAsync(db.CurrentTenantId!.Value) ?? throw new ApiException(404, "Firma bulunamadı.");
        if (!string.IsNullOrWhiteSpace(dto.Name)) t.Name = dto.Name;
        t.Phone = dto.Phone; t.Email = dto.Email; t.Address = dto.Address;
        t.TaxNumber = dto.TaxNumber; t.TaxOffice = dto.TaxOffice; t.UpdatedAt = DateTime.UtcNow;
        await db.SaveChangesAsync();
        return Ok(new { message = "Firma bilgileri güncellendi." });
    }
}

File: api/app/Http/Controllers/Api/SettingsController.php
<?php
// app/Http/Controllers/Api/SettingsController.php

namespace App\Http\Controllers\Api;

use App\Http\Controllers\Controller;
use Illuminate\Http\Request;

class SettingsController extends Controller
{
    /** Firma bilgileri + abonelik durumu. */
    public function show(Request $request)
    {
        $tenant = $request->user()->tenant;

        return response()->json([
            'name'        => $tenant->name,
            'slug'        => $tenant->slug,
            'phone'       => $tenant->phone,
            'email'       => $tenant->email,
            'address'     => $tenant->address,
            'tax_number'  => $tenant->tax_number,
            'tax_office'  => $tenant->tax_office,
            'logo_url'    => $tenant->logo_url,
            'plan'        => $tenant->plan,
            'plan_expires_at' => $tenant->plan_expires_at,
            'sms_balance' => $tenant->sms_balance,
        ]);
    }

    /** Firma bilgilerini güncelle (yalnızca yönetici). */
    public function update(Request $request)
    {
        $data = $request->validate([
            'name'       => ['sometimes', 'string', 'max:255'],
            'phone'      => ['nullable', 'string', 'max:20'],
            'email'      => ['nullable', 'email', 'max:255'],
            'address'    => ['nullable', 'string'],
            'tax_number' => ['nullable', 'string', 'max:20'],
            'tax_office' => ['nullable', 'string', 'max:100'],
        ]);

        $tenant = $request->user()->tenant;
        $tenant->update($data);

        return response()->json(['message' => 'Firma bilgileri güncellendi.']);
    }
}
